### PR TITLE
correct calculation of sum_self_cuda_time_total

### PR DESCRIPTION
**Pull request by youkaichao**
it seems the calculation of `sum_self_cuda_time_total` is wrong and misses much time.

Following the tutorial in https://pytorch.org/tutorials/recipes/recipes/profiler_recipe.html , I found that percentages of "Self CUDA %" sum up to over 100%, which means there must be some missing cuda time.

The profiler code assumes no cuda time if `evt.is_legacy == False`. However, it is not the case. I print some message here, showing that `evt.self_cuda_time_total` can be non-zero even if `evt.is_legacy == False`:

```
debug: evt.key:aten::convolution_backward evt.self_cuda_time_total:1082941, evt.is_legacy:False
debug: evt.key:aten::cudnn_convolution evt.self_cuda_time_total:380612, evt.is_legacy:False
debug: evt.key:aten::cudnn_batch_norm_backward evt.self_cuda_time_total:274686, evt.is_legacy:False
debug: evt.key:aten::cudnn_batch_norm evt.self_cuda_time_total:177172, evt.is_legacy:False
debug: evt.key:cudaLaunchKernel evt.self_cuda_time_total:140516, evt.is_legacy:False
debug: evt.key:aten::threshold_backward evt.self_cuda_time_total:135466, evt.is_legacy:False
debug: evt.key:aten::clamp_min_ evt.self_cuda_time_total:92828, evt.is_legacy:False
debug: evt.key:aten::add_ evt.self_cuda_time_total:86217, evt.is_legacy:False
debug: evt.key:aten::add evt.self_cuda_time_total:78447, evt.is_legacy:False
debug: evt.key:cudaDeviceGetStreamPriorityRange evt.self_cuda_time_total:60155, evt.is_legacy:False
debug: evt.key:cudaStreamGetPriority evt.self_cuda_time_total:55187, evt.is_legacy:False
debug: evt.key:cudaStreamIsCapturing evt.self_cuda_time_total:50766, evt.is_legacy:False
debug: evt.key:cudaOccupancyMaxActiveBlocksPerMultiprocessorWithFlags evt.self_cuda_time_total:41838, evt.is_legacy:False
debug: evt.key:cudaMemsetAsync evt.self_cuda_time_total:20459, evt.is_legacy:False
debug: evt.key:aten::max_pool2d_with_indices_backward evt.self_cuda_time_total:20100, evt.is_legacy:False
debug: evt.key:cudaFree evt.self_cuda_time_total:7470, evt.is_legacy:False
debug: evt.key:aten::max_pool2d_with_indices evt.self_cuda_time_total:6777, evt.is_legacy:False
......
```

Therefore, the correct sum of self_cuda_time_total should be calculated regardless of `evt.is_legacy`. After the correction, the percentages of "Self CUDA %" sum up to about 100%.

cc @robieta @chaekit @aaronenyeshi @ngimel @nbcsm @guotuofeng @guyang3532 @gaoteng-git @tiffzhaofb @dzhulgakov

### CODE REVIEW

**Comment by diana-pwf** [comment]
I ran into this issue, tooo! Hope it would be fixed early!

**Comment by ngimel** [comment]
Duplicate of #60783, cc @davidberard98

**Review by ngimel — APPROVED** [review]

**Comment by aaronenyeshi** [comment]
Just wanted to clarify my understanding of the issue here.

In the non-legacy path, a kernel's self CUDA time is attributed to both the CUDA event and the Operator event. So if we land this change, the sum_self_cuda_time_total will be ~2x of the actual sum of Self CUDA time. 

Ex. A CPU op, `aten::_softmax` launches a `softmax_warp_forward` GPU kernel, and a CPU op, `aten::copy_` launches 2 `at::native:unrolled` GPU kernels. If `softmax_warp_forward` takes 100us and `at::native::unrolled` takes 25us, the Self CUDA would be:
 - `aten::_softmax` 100us
 - `softmax_warp_forward` 100us
 - `aten::copy_` 50us
 - `at::native_unrolled` 50us

The total CUDA time should be 150us, but if we sum all the events up, it would be 300us (double the actual). Percentage-wise, Self CUDA % will add up to 100%, but may be false. Ie, softmax shows as 33% instead of 66%.

Is this the correct understanding?
